stage1/init/pod.go

```diff
@@ -486,7 +487,14 @@ func (p *Pod) appToNspawnArgs(ra *schema.RuntimeApp) ([]string, error) {
 	for _, v := range p.Manifest.Volumes {
 		vols[v.Name] = v
 		if v.Kind == "empty" {
-			if err := os.MkdirAll(filepath.Join(common.SharedVolumesPath(p.Root), v.Name.String()), 0755); err != nil {
+			sharedVolPath := common.SharedVolumesPath(p.Root)
+			if err := os.MkdirAll(sharedVolPath, sharedVolPerm); err != nil {
+				return nil, fmt.Errorf("could not create shared volumes directory: %v", err)
+			}
+			if err := os.Chmod(sharedVolPath, sharedVolPerm); err != nil {
+				return nil, fmt.Errorf("could not change permissions of %q: %v", sharedVolPath, err)
+			}
```

Comment:
You could move this `SharedVolumesPath()`, `MkdirAll()` and `Chmod()` before the `for` loop, since it's only needed to be executed one time? Or you could let the `MkdirAll(Join())` create the parent directory in the loop and move the `Chmod(sharedVolPath, sharedVolPerm)` after the loop (if at least 1 empty volume exists).